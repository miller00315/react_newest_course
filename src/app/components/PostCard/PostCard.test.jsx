import { render, screen } from '@testing-library/react';
import {PostCard} from './index'

const postMock = {
    title: 'testTitle',
    body: 'testBody',
    id: 1,
    cover: 'img/img.png'
}

describe('<PostCard />', () => {
    it('Should render a PostCard correctly', () => {
        render(<PostCard post={postMock} />);

        expect(screen.getByRole('img', {name: 'testTitle'})).toHaveAttribute('src', postMock.cover);

        expect(screen.getByText('testTitle')).toBeInTheDocument();

        expect(screen.getByText('testBody')).toBeInTheDocument();
    });

    it('Should match a snapshot', () => {
      const { container } = render(<PostCard post={postMock} />);

      expect(container.firstChild).toMatchSnapshot();
    });
});