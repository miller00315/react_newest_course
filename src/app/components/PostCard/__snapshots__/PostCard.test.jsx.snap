// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match a snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="testTitle"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      testTitle
    </h2>
    <p>
      testBody
    </p>
  </div>
</div>
`;
